doc/plutus/howtos/plutus-spending-script-example.rst

```diff
@@ -0,0 +1,171 @@
+
+- the creation of the ``AlwaysSucceeds`` Plutus txin script
+- sending ADA to the Plutus script address
+- spending ADA at the Plutus script address
+
+In this example we will use the
+`AlwaysSucceeds <../plutus-example/plutus-example/src/Cardano/PlutusExample/AlwaysSucceeds.hs>`__
+Plutus spending script. In order to execute a Plutus spending script, we
+require the following:
+
+- Collateral tx input(s) - these are provided and are forfeited in the
+  event the Plutus script fails to execute.
+- A Plutus tx output with accompanying datum hash. This is the tx
+  output that sits at the Plutus script address. It must have a datum
+  hash, otherwise, it is unspendable.
+- The Plutus script serialized in the text envelope format.
+  ``cardano-cli`` expects Plutus scripts to be serialized in the text
+  envelope format.
+
+Creating the ``AlwaysSucceeds`` Plutus spending script
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The plutus-example executable will automagically generate several Plutus
+scripts in the CLI-compatiable text envelope format.
+
+Run the following commands:
+
+.. code:: bash
+
+   cd plutus-example
+
+   cabal run exe:plutus-example
+
+This will output ``always-succeeds-txin.plutus`` in the
+``generated-plutus-scripts`` dir.
+
+Setting up a local Alonzo node cluster
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There is a convenient script that will set up an Alonzo cluster
+immediately on your local machine.
+
+Run the following command:
+
+.. code:: bash
+
+   cabal install cardano-cli
+   cabal install cardano-node
+   ./scripts/byron-to-alonzo/mkfiles.sh alonzo
+
+Follow the instructions displayed in the terminal to start your Alonzo
+cluster.
+
+Sending ADA to the script address
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to require a Plutus script to validate the spending of a tx
+ouput, we must put the tx output at the script address of the said
+Plutus script. However, before we do that, we must create a datum hash:
+
+.. code:: bash
+
+   > cardano-cli transaction hash-script-data --script-data-value 42
+   > 9e1199a988ba72ffd6e9c269cadb3b53b5f360ff99f112d9b2ee30c4d74ad88b
+
+In this example, the script we are using always succeeds so we can use
+any datum hash. We calculate the script address as follows:
+
+.. code:: bash
+
+   > cardano-cli address build --payment-script-file plutus-example/generated-plutus-scripts/always-succeeds-txin.plutus  --testnet-magic 42
+   > addr_test1wzeqkp6ne3xm6gz39l874va4ujgl4kr0e46pf3ey8xsu3jsgkpcj2
+
+Now, we should create the tx that will send ADA to the script address of
+our ``AlwaysSucceeds`` script:
+
+.. code:: bash
+
+   cardano-cli transaction build-raw \
+     --alonzo-era \
+     --fee 0 \
+     --tx-in $txin \
+     --tx-out "addr_test1wzeqkp6ne3xm6gz39l874va4ujgl4kr0e46pf3ey8xsu3jsgkpcj2+$lovelace" \
+     --tx-out-datum-hash 9e1199a988ba72ffd6e9c269cadb3b53b5f360ff99f112d9b2ee30c4d74ad88b \
+     --out-file create-datum-output.body
```

Comment:
`$txin` does reference anything?

Reply:
Same for some other variables mentioned in this file.

Reply:
I linked to this script to see how variables are assigned: https://github.com/input-output-hk/cardano-node/blob/master/scripts/plutus/example-txin-locking-plutus-script.sh